File: 04.02.最小高度树/04.02.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sortedArrayToBST(nums []int) *TreeNode {
    if len(nums) == 0 {
        return nil
    }
    if len(nums) == 1 {
        return &TreeNode{Val: nums[0]}
    }

    mid := len(nums) / 2
    head := &TreeNode{
        Val: nums[mid],
    }
    head.Left = sortedArrayToBST(nums[:mid])
    head.Right = sortedArrayToBST(nums[mid+1:])
    return head
}

File: 04.02.最小高度树/04.02.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* dfs(const vector<int> &nums, int L, int R) {
        if(L > R) {
            return nullptr;
        }
        int mid = (L+R)>>1;
        auto ptr = new TreeNode(nums[mid]);
        ptr->left = dfs(nums, L, mid-1);
        ptr->right = dfs(nums, mid+1, R);
        return ptr;
    }
    TreeNode* sortedArrayToBST(vector<int>& nums) {
        return dfs(nums, 0, nums.size()-1);
    }
};